src/assets/theme/ThemeProvider.tsx

```diff
@@ -0,0 +1,25 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import React, { ReactElement, ReactChild, ReactChildren } from "react";
+import { ThemeProvider } from "styled-components";
+import { connect } from "react-redux";
+
+import { ApplicationStateModel } from "../../state/application-state.model";
+import { ThemeKey, themes } from ".";
+
+interface TPR {
+  themeKey: ThemeKey;
+  children: ReactChild | ReactChildren | ReactElement;
```

Comment:
Teraz ten typ chyba nie obsługuje stringów, booleanów, listy elementów i innych podobnych do tego rzeczy.

Może lepiej zamiast tego użyć `ReactNode`?

Wiem, że to raczej nie jest ważne w kontekście tego komponentu, ale można by było uprościć kod w taki sposób.

Chyba że to że nie uwzględniamy tych rzeczy było zamierzone.


[Tu](https://stackoverflow.com/a/58123882/7516134)  jak by coś można trochę więcej na ten temat poczytać

Reply:
ok